python/ray/air/execution/resources/virtual.py

```diff
@@ -0,0 +1,24 @@
+from typing import Dict, List, Union
+
+from dataclasses import dataclass
+
+import ray
+from ray.air.execution.resources.fixed import FixedResourceManager
+from ray.air.execution.resources.request import AllocatedResource
+from ray.util.annotations import DeveloperAPI
+
+
+@DeveloperAPI
+@dataclass
+class VirtualAllocatedResource(AllocatedResource):
+    bundles: List[Dict[str, float]]
+
+    def annotate_remote_objects(
+        self, objects
+    ) -> List[Union[ray.ObjectRef, ray.actor.ActorHandle]]:
+        return objects
+
+
+@DeveloperAPI
+class VirtualResourceManager(FixedResourceManager):
```

Comment:
what is this for?

Reply:
It reserves resources but does not bind the resources to the remote objects. If I recall correctly it was needed for some use cases of the prototype (I think for some nesting cases)